qiskit_experiments/library/characterization/analysis/stark_ramsey_fast_analysis.py

```diff
@@ -0,0 +1,303 @@
+
+        options = super()._default_options()
+        options.update_options(
+            result_parameters=[
+                curve.ParameterRepr("c1_pos", "stark_pos_coef_o1", "Hz"),
+                curve.ParameterRepr("c2_pos", "stark_pos_coef_o2", "Hz"),
+                curve.ParameterRepr("c3_pos", "stark_pos_coef_o3", "Hz"),
+                curve.ParameterRepr("c1_neg", "stark_neg_coef_o1", "Hz"),
+                curve.ParameterRepr("c2_neg", "stark_neg_coef_o2", "Hz"),
+                curve.ParameterRepr("c3_neg", "stark_neg_coef_o3", "Hz"),
+                curve.ParameterRepr("f_err", "stark_ferr", "Hz"),
+            ],
+            data_subfit_map={
+                "Xpos": {"series": "X", "direction": "pos"},
+                "Ypos": {"series": "Y", "direction": "pos"},
+                "Xneg": {"series": "X", "direction": "neg"},
+                "Yneg": {"series": "Y", "direction": "neg"},
+            },
+            plotter=ramsey_plotter,
+        )
+
+        return options
+
+    def _generate_fit_guesses(
+        self,
+        user_opt: curve.FitOptions,
+        curve_data: curve.CurveData,
+    ) -> Union[curve.FitOptions, List[curve.FitOptions]]:
+        """Create algorithmic initial fit guess from analysis options and curve data.
+
+        Args:
+            user_opt: Fit options filled with user provided guess and bounds.
+            curve_data: Formatted data collection to fit.
+
+        Returns:
+            List of fit options that are passed to the fitter function.
+        """
+        # Compute offset guess
+        user_opt.p0.set_if_empty(
+            offset=np.mean(curve_data.y),
+            f_err=0.0,
+        )
+        user_opt.bounds.set_if_empty(
+            offset=(-1, 1),
+            amp=(0, 1),
+        )
+        est_offs = user_opt.p0["offset"]
+
+        # Compute amplitude guess
+        amps = np.zeros(0)
+        for direction in ("pos", "neg"):
+            ram_x_off = curve_data.get_subset_of(f"X{direction}").y - est_offs
+            ram_y_off = curve_data.get_subset_of(f"Y{direction}").y - est_offs
+            amps = np.concatenate([amps, np.sqrt(ram_x_off**2 + ram_y_off**2)])
+        user_opt.p0.set_if_empty(amp=np.median(amps))
+        est_a = user_opt.p0["amp"]
+        d_const = user_opt.p0["dt"]
+
+        # Compute polynominal coefficients
+        guesses = []
+        for direction in ("pos", "neg"):
+            ram_x_data = curve_data.get_subset_of(f"X{direction}")
+            ram_y_data = curve_data.get_subset_of(f"Y{direction}")
+            xvals = ram_x_data.x
+
+            # Get normalized sinusoidals
+            xnorm = (ram_x_data.y - est_offs) / est_a
+            ynorm = (ram_y_data.y - est_offs) / est_a
+
+            # Compute derivative to extract polynominals from sinusoidal
+            dx = np.diff(xnorm) / np.diff(xvals)
+            dy = np.diff(ynorm) / np.diff(xvals)
```

Comment:
Minor: Would it help to do a more sophisticated derivative, maybe with some smoothing?

Reply:
I was also thinking of it, but this requires more data points. Experimentalists want to characterize the shift within full AWG range to detect more TLS in as wider frequency as possible, but such high amp will cause rapid oscillation and smoothing with something like SG filter drastically distorts the signal due to LPF effect.